Sources_v3/Controllers/CurrentUserController/CurrentUserController.swift

```diff
@@ -100,6 +100,121 @@ public class CurrentUserControllerGeneric<ExtraData: ExtraDataTypes>: Controller
     }
 }
 
+// MARK: - Set current user
+
+public extension CurrentUserControllerGeneric {
+    /// Sets a new anonymous as a current user.
+    ///
+    /// Anonymous users have limited set of permissions. A typical use case for anonymous users are livestream channels,
+    /// where they are allowed to read the conversation.
+    ///
+    /// - Parameters:
+    ///   - completion: Called when the new anonymous user is set. If setting up the new user fails, the completion
+    /// is called with an error.
+    func setAnonymousUser(completion: ((Error?) -> Void)? = nil) {
+        client.disconnect()
+        client.prepareEnvironmentForNewUser(userId: .anonymous, role: .anonymous, extraData: nil) { error in
+            guard error == nil else {
+                completion?(error)
+                return
+            }
+            
+            self.client.connect(completion: completion)
+        }
+    }
+    
+    /// Sets a new **guest** user as a current user.
+    ///
+    /// Guest sessions do not require any server-side authentication.
+    /// Guest users have a limited set of permissions.
+    ///
+    /// - Parameters:
+    ///   - userId: The new guest-user identifier.
+    ///   - extraData: The extra data of the new guest-user.
+    ///   - completion: The completion. Will be called when the new guest user is set.
+    ///                 If setting up the new user fails the completion will be called with an error.
+    func setGuestUser(userId: UserId, extraData: ExtraData.User, completion: ((Error?) -> Void)? = nil) {
+        client.disconnect()
+        client.prepareEnvironmentForNewUser(userId: userId, role: .guest, extraData: extraData) { error in
+            guard error == nil else {
+                completion?(error)
+                return
+            }
+            
+            self.client.apiClient.request(endpoint: .guestUserToken(userId: userId, extraData: extraData)) {
```

Comment:
Previously we did `[weak self]` but I think it's not the best decision:
1. once we set a new user I'd expect it to be finished regardless of a **CurrentUserControllerGeneric** lifecycle
2. as a developer I want be able to do `client.currentUserController.setUser(...)` and don't care about maintaining a strong ref to **CurrentUserControllerGeneric**

Reply:
Yep, makes total sense 👍